server/pulp/server/db/model/__init__.py

```diff
@@ -631,7 +631,7 @@ def unit_key_str(self):
         """
         The unit key represented as a string ordered by unit key fields alphabetically
         """
-        return str(sorted([getattr(self, key) for key in self.unit_key_fields]))
+        return str(sorted(self.unit_key.items()))
```

Comment:
I wonder if depending on the string representation of a list of tuples could turn out to be non-portable. What if it changes subtly in python 3.x, for example? Since this change is the first time we're expecting results of this to persist long-term, maybe we should consider how to make it more deterministic.

An easy solution is to choose a string that's unlikely to be part of a unit key value and use it as a separator. The hard part is choosing that string. Something like this might work, although I'm at least half-kidding about the example separator:

```
return '!@#$%^&*'.join(self.unit_key[key] for key in sorted(self.unit_key.keys()))
```

What do you think? Maybe there's a better way.